src/replay/service.zig

```diff
@@ -148,16 +155,18 @@ pub const Receivers = struct {
 
 const ReplayState = struct {
     allocator: Allocator,
+    my_identity: Pubkey,
     logger: Logger,
+    thread_pool: *ThreadPool,
     slot_leaders: SlotLeaders,
+    slot_tracker: *SlotTracker,
+    epochs: *EpochTracker,
+    hard_forks: sig.core.HardForks,
+    account_store: AccountStore,
+    progress_map: *ProgressMap,
     ledger: LedgerRef,
-    account_store: sig.accounts_db.AccountStore,
-
-    my_identity: Pubkey,
+    execution: ReplayExecutionState,
```

Comment:
It appears many changes I made to the replaystate struct were reverted here. This is supposed to be returned by a method instead, making this replaystruct state able to embed many fields directly without being self-referential.